src/app/server/CommissioningWindowManager.cpp

```diff
@@ -306,11 +306,13 @@ CHIP_ERROR CommissioningWindowManager::OpenBasicCommissioningWindow(Seconds32 co
                                                                     CommissioningWindowAdvertisement advertisementMode)
 {
     RestoreDiscriminator();
+    bool isBle = false;
 
 #if CONFIG_NETWORK_LAYER_BLE
     // Enable BLE advertisements if commissioning window is to be opened on all supported
     // transports, and BLE is supported on the current device.
-    SetBLE(advertisementMode == chip::CommissioningWindowAdvertisement::kAllSupported);
+    isBle = advertisementMode == chip::CommissioningWindowAdvertisement::kAllSupported;
+    SetBLE(isBle);
```

Comment:
So... "Extended Announcement" is not limited to BLE, as far as I can tell.  All it requires is that the device be "uncommissioned".

So it seems like that is what we should be implementing here: when asked to open a basic commissioning window, if we are not already commissioned, allow extended announcement duration.  In all other cases, disallow it.  No relationship to BLE.

Reply:
You are right, thanks for catching it. In such case the `CHIP_DEVICE_CONFIG_BLE_EXT_ADVERTISING` define sounds a little bit misleading. I will update it in this PR by removing the BLE part.

Reply:
Another thing is that extended announcement seems to not be 100% ready for the IP transport, as in the dnssd implementation the VID and PID are always added to the commissioning data. Probably it should work similarly to BLE and update advertised data after exceeding the timeout and entering the extended part.

Reply:
Yes, that last bit needs to be fixed if we're going to support this at all.  Is there an issue tracking that?

Reply:
I've created one: https://github.com/project-chip/connectedhomeip/issues/35592